src/main/java/org/radarcns/hdfs/Application.java

```diff
@@ -193,10 +203,41 @@ public void start() {
         System.setProperty("java.util.concurrent.ForkJoinPool.common.parallelism",
                 String.valueOf(settings.getNumThreads() - 1));
 
-        Instant timeStart = Instant.now();
-        RadarHdfsRestructure hdfsReader = new RadarHdfsRestructure(this);
         try {
             Files.createDirectories(settings.getTempDir());
+        } catch (IOException ex) {
+            logger.error("Failed to create temporary directory");
+            return;
+        }
+
+        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+        executorService.execute(() -> hdfsReader = new RadarHdfsRestructure(this));
+
+        if (isService) {
+            logger.info("Press Ctrl+C to exit...");
+            executorService.scheduleAtFixedRate(this::runRestructure,
+                    pollInterval / 4, pollInterval, TimeUnit.SECONDS);
```

Comment:
What happens if a restructure is already running ? it will start another one after poll interval ? Should we add a locking mechanism as well?

Reply:
It's running a single thread executor, so the restructure scripts will not overlap.

Reply:
We considered using a locking mechanism instead, so we can also scale this service up, but that would take more significant changes. See #48.